.github/workflows/node.js.yml

```diff
@@ -8,22 +8,70 @@ on:
   workflow_call:
 
 jobs:
+  
   build:
-
     runs-on: ubuntu-latest
-
+    outputs:
+      coverage: ${{ steps.coverage.outputs.coverage }}
+      color: ${{ steps.coverage.outputs.color }}
     strategy:
       matrix:
         node-version: [16.x, 18.x]
         # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
-
     steps:
     - uses: actions/checkout@v3
     - name: Use Node.js ${{ matrix.node-version }}
       uses: actions/setup-node@v3
       with:
         node-version: ${{ matrix.node-version }}
         cache: 'npm'
+    - name: Checkout code
+      uses: actions/checkout@v2
     - run: npm ci
     - run: npm run fmt:check
-    - run: npm run test:ci
+    - id: coverage
+      run: |
+        value=$(npm run test:ci | grep -oE '^│\s*total\s+\│\s*([0-9.]+)%' | grep -oE '[0-9]+(\.[0-9]+)?%' | cut -d'.' -f1)
+        echo "coverage=$value" >> $GITHUB_OUTPUT
+
+    - name: Update README
+      uses: actions/github-script@v3
+      with:
+        script: |
+          const color = process.env.COVERAGE >= 70 ? `green` : `orange`;
```

Comment:
I propose 3 colors:
```
- >= 80: green
- > 70: orange
- else red
```

Reply:
But if you make it generic, maybe we will need make it updatable by the caller.

Reply:
> I propose 3 colors:
> 
> ```
> - >= 80: green
> - > 70: orange
> - else red
> ```

sure we can do this

Reply:
> 
When you say make it updatable by the caller, you mean update the ranges ?